### PR TITLE
Use `julia-mirror` to mirror Julia ecosystem

### PR DESCRIPTION
**Pull request by staticfloat**
### Checklist

- [X] 更新 README

This uses the [`julia-mirror` scripts](https://github.com/sunoru/julia-mirror) to download the Julia ecosystem (Julia versions, Julia packages, etc...) to `/data` within the docker container, outputting logs into `/logs`.

### CODE REVIEW

File: julia/sync.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (zhsj) [inline julia/sync.sh]: Add change the shabang.
```suggestion
#!/bin/sh
```

File: julia/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM ustcmirror/base:alpine
+MAINTAINER Elliot Saba <staticfloat@gmail.com>
+RUN apk add --no-cache python3 git bash && \
```
Comment (zhsj) [inline julia/Dockerfile]: Is it a hard dependency for bash? If not, probably just use busybox's shell

```suggestion
RUN apk add --no-cache python3 git && \
```

**Comment by staticfloat** [comment]
You are correct of course; doesn't need `bash`.  I had that for debugging and forgot to remove it.  :)

**Comment by gaoyifan** [comment]
Actually, we add bash in base image 😂
https://github.com/ustclug/ustcmirror-images/blob/master/base/Dockerfile.alpine#L7

File: julia/sync.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+mkdir -p /data/tmp
```
Comment (gaoyifan) [inline julia/sync.sh]: Using `$TO` instead fixed path `/data` is better. Refer to [README.md](https://github.com/ustclug/ustcmirror-images#volumes)
```suggestion
mkdir -p $TO/tmp
```

File: julia/sync.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+python3 /julia-mirror/scripts/mirror_julia.py /data --logging-file /logs/mirror_julia.log --temp-dir=/data/tmp
```
Comment (gaoyifan) [inline julia/sync.sh]: `/data` -> `$TO`

File: julia/sync.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+rm -rf /data/tmp
```
Comment (gaoyifan) [inline julia/sync.sh]: `/data` -> `$TO`

File: julia/sync.sh (hunk 3 above)
Comment (staticfloat) [reply julia/sync.sh]: Ah, I didn't understand `README.md`.  Thanks for the tip.

File: julia/sync.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+mkdir -p ${TO}/tmp
+python3 /julia-mirror/scripts/mirror_julia.py ${TO} --logging-file ${LOGDIR}/mirror_julia.log --temp-dir=${TO}/tmp
```
Comment (knight42) [inline julia/sync.sh]: It's recommended that the program outputs all messages to stdout.

Comment (staticfloat) [reply julia/sync.sh]: Does that mean you do not want me to use a log file, or that you would prefer that it output to both?

Comment (gloit042) [reply julia/sync.sh]: `ustcmirror` automatically collect logs from outputs of scripts.

Comment (knight42) [reply julia/sync.sh]: @staticfloat You don't need to log to files explicitly, both stdout and stderr would be captured by the [entrypoint](https://github.com/ustclug/ustcmirror-images/blob/master/base/entry.sh#L70).

**Comment by staticfloat** [comment]
Awesome, thanks guys!

**Pull request merged**